lockfile/src/lib.rs

```diff
@@ -249,7 +250,8 @@ pub fn get_depfile_path_format<P: AsRef<Path>>(path: P) -> Option<LockfileFormat
     let path = path.as_ref();
     LockfileFormat::iter().find(|format| {
         let parser = format.parser();
-        parser.is_path_lockfile(path) || parser.is_path_manifest(path)
+        format != &LockfileFormat::GoMod
+            && (parser.is_path_lockfile(path) || parser.is_path_manifest(path))
```

Comment:
Same comment as the path format function. It's still a valid format, it just shouldn't be suggested together with other go lockfiles.

Reply:
Removed